File: cost/iona/orbacus/cpp/poa/TestLocationForward_impl.cpp
// **********************************************************************
//
// Copyright (c) 2001
// Object Oriented Concepts, Inc.
// Billerica, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

#include <includes.h>

#include <TestLocationForwardActivator_skel.h>

#include <Test_impl.h>
#include <TestLocationForward_impl.h>

using namespace CORBA;
using namespace PortableServer;

TestLocationForward_impl::TestLocationForward_impl(
    ORB_ptr orb,
    POA_ptr poa)
    : Test_impl(orb, poa)
{
}

void
TestLocationForward_impl::deactivate_servant()
{
    ObjectId_var oid;
    POA_var poa;

    try
    {
	oid = current_ -> get_object_id();
	poa = current_ -> get_POA();
    }
    catch(const PortableServer::Current::NoContext&)
    {
	assert(false);
    }
    
    try
    {
	poa -> deactivate_object(oid);
    }
    catch(const POA::WrongPolicy&)
    {
	assert(false);
    }
    catch(const POA::ObjectNotActive&)
    {
	assert(false);
    }
}

File: cost/legacy/iona/orbacus/java/poa/TestLocationForward_impl.java
// **********************************************************************
//
// Copyright (c) 2000
// Object Oriented Concepts, Inc.
// Billerica, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

package test.poa;

import org.omg.CORBA.*;
import org.omg.PortableServer.*;
import org.omg.PortableServer.POAPackage.*;

final class TestLocationForward_impl extends TestLocationForwardPOA
{
    private Test_impl delegate_;

    TestLocationForward_impl(ORB orb)
    {
	delegate_ = new Test_impl(orb, "", false);
    }

    public void
    deactivate_servant()
    {
	byte[] oid = null;
	POA poa = null;

	try
	{
	    oid = delegate_.current_.get_object_id();
	    poa = delegate_.current_.get_POA();
	}
	catch(org.omg.PortableServer.CurrentPackage.NoContext ex)
	{
	    throw new RuntimeException();
	}

	try
	{
	    poa.deactivate_object(oid);
	}
	catch(WrongPolicy ex)
	{
	    throw new RuntimeException();
	}
	catch(ObjectNotActive ex)
	{
	    throw new RuntimeException();
	}
    }

    public void
    aMethod()
    {
	delegate_.aMethod();
    }
}
